sdcflows/interfaces/fmap.py

```diff
@@ -23,6 +24,111 @@
+        se_rescaled_output = fname_presuffix(short_echo_image, suffix='_shortTE_scaled',
+                                             newpath=runtime.cwd)
+        nb.Nifti1Image(se_rescaled_data, se_phase_image.affine, se_phase_image.header
+                       ).to_filename(se_rescaled_output)
+        self._results['short_te_phase_image'] = se_rescaled_output
+        self._results['short_te_phase_metadata'] = metadatas[short_echo_index]
+
+        # Rescale and save the long TE image
+        le_phase_image = nb.load(long_echo_image)
+        le_phase_data = le_phase_image.get_fdata()
+        le_rescaled_data = rescale_phase_image(le_phase_data)
+        le_rescaled_output = fname_presuffix(long_echo_image, suffix='_longTE_scaled',
+                                             newpath=runtime.cwd)
+        nb.Nifti1Image(le_rescaled_data, le_phase_image.affine, le_phase_image.header
+                       ).to_filename(le_rescaled_output)
+        self._results['long_te_phase_image'] = le_rescaled_output
+        self._results['long_te_phase_metadata'] = metadatas[long_echo_index]
+
+        merged_metadata = deepcopy(metadatas[0])
+        del merged_metadata['EchoTime']
+        merged_metadata['EchoTime1'] = float(echo_times[short_echo_index])
+        merged_metadata['EchoTime2'] = float(echo_times[long_echo_index])
+        self._results['phasediff_metadata'] = merged_metadata
+        return runtime
+
+
+def rescale_phase_image(phase_data):
+    """Ensure that phase images are in a usable range for unwrapping.
+
+    From the FUGUE User guide::
+
+        If you have seperate phase volumes that are in integer format then do:
+
+        fslmaths orig_phase0 -mul 3.14159 -div 2048 phase0_rad -odt float
+        fslmaths orig_phase1 -mul 3.14159 -div 2048 phase1_rad -odt float
+
+        Note that the value of 2048 needs to be adjusted for each different
+        site/scanner/sequence in order to be correct. The final range of the
+        phase0_rad image should be approximately 0 to 6.28. If this is not the
+        case then this scaling is wrong. If you have separate phase volumes are
+        not in integer format, you must still check that the units are in radians,
+        and if not scale them appropriately using fslmaths.
+    """
+    imax = phase_data.max()
+    imin = phase_data.min()
+    scaled = (phase_data - imin) / (imax - imin)
+    return 2 * np.pi * scaled
+
+
+class PhaseDiffInputSpec(BaseInterfaceInputSpec):
+    phasediff = File(exists=True, desc='single-volume phasediff image')
+    phase1 = File(exists=True, desc='single-volume phase image')
+    phase2 = File(exists=True, desc='single-volume phase image')
+    metadata = traits.Dict(mandatory=True)
+
+
+class PhaseDiffOutputSpec(TraitedSpec):
+    phasediff = File(exists=True, mandatory=True)
+
+
+class PhaseDiff(SimpleInterface):
+    input_spec = PhaseDiffInputSpec
+    output_spec = PhaseDiffOutputSpec
+
```

Comment:
Isn't this missing the implementation?

Reply:
I deleted this class because the phasediff is calculated in a workflow ((https://github.com/mattcieslak/sdcflows/blob/phase1phase2/sdcflows/workflows/phdiff.py#L30) instead. This was preferable to calling PRELUDE a bunch of times inside the interface.

Regarding @tsalo's suggestion - this is implemented in init_calculate_phdiff_wf using 3dcalc.

Reply:
Okay, if I understand, lines 119-133 should then go away too? 
https://github.com/poldracklab/sdcflows/pull/30/files#diff-422ff9092ea3510c48e30863f876a550R119-R133